src/xsdba/_adjustment.py

```diff
@@ -135,18 +135,19 @@ def dqm_train(
     `jitter_over_thresh_value` and `jitter_over_thresh_upper_bnd` must be both be specified to
     use `jitter_over_thresh`, or both be None (default) to skip it.
     """
+    dim = [dim] if isinstance(dim, str) else dim
+    # Ensure we only reduce on valid dims, allows for extra dims like "realization" on the sim
+    ref_dim = Grouper.filter_dim(ds.ref, dim)
+    sim_dim = Grouper.filter_dim(ds.hist, dim)
     ds = _preprocess_dataset(
         ds,
-        dim,
+        ref_dim,
```

Comment:
the idea behind this, and I'm not sure it's 100% OK: 

If `sim` has extra `ADD_DIMS`, then they will not be included. This assumes that it's always `sim` which can have a superset of dimensions of `ref`